boards/arm/nrf51_ble400/board.h

```diff
@@ -0,0 +1,38 @@
+/*
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#ifndef __INC_BOARD_H
+#define __INC_BOARD_H
+
+#include <soc.h>
+
+/* Push button switch 0 KEY1 */
+#define SW0_GPIO_PIN	16
+#define SW0_GPIO_NAME	CONFIG_GPIO_NRF5_P0_DEV_NAME
+
```

Comment:
you may want PULL UP settings? look at https://github.com/zephyrproject-rtos/zephyr/pull/6415

Reply:
sorry, not sure I understand what you mean

Reply:
ohh - got it. is it mandatory? really my samples was work well without it and btw other nrf51 boards has no PULL UP settings.

Reply:
Have you run the button sample? does the buttons have external pull up resisters? if not, you will need the GPIO driver to configure internal pull ups.

Reply:
ok will check

Reply:
Yes I got button sample working (I can provide video if required). But anyway will check if board has external pull ups.

Reply:
@anangl can you confirm if nRF51 needs PULL up configuration or it is for sure default PULL UP?

Reply:
Hmm according schematics there are no pull ups https://www.waveshare.com/w/upload/1/1b/BLE400-Schematic.pdf

Reply:
Added pull ups and checked button sample again. It work. Compared difference with this options on/off - when pull up off I see more false key presses. So you are right - pull up required here.